spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/DenseTensor.scala

```diff
@@ -1717,24 +1720,64 @@ private[tensor] class DenseTensor[@specialized T: ClassTag](
     val indicesTensor = if (indices == null) Tensor[T]() else indices
     indicesTensor.resize(topKSize)
 
+    @inline
+    def compare(a: T, b: T): Boolean = if (increase) {
+      ev.isGreater(b, a)
+    } else {
+      ev.isGreater(a, b)
+    }
+
+    class LocalComparator(increase: Boolean) extends Comparator[(T, Int)] {
+      override def compare(o1: (T, Int), o2: (T, Int)): Int = {
+        val ret = if (ev.isGreater(o1._1, o2._1)) {
```

Comment:
why do we need this `if` as it's covered by below `if`

Reply:
Because there's no `equal` in ev. The next `isGreaterEq` handles the `equal`.